hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/util/DistCpUtils.java

```diff
@@ -592,10 +592,14 @@ public static CopyMapper.ChecksumComparison checksumsAreEqual(
     // comparison that took place and return not compatible.
     // else if matched, return compatible with the matched result.
     if (sourceChecksum == null || targetChecksum == null) {
+      LOG.error("Checksum incompatible. Source checksum: {}, target checksum: {}",
+          sourceChecksum, targetChecksum);
       return CopyMapper.ChecksumComparison.INCOMPATIBLE;
     } else if (sourceChecksum.equals(targetChecksum)) {
       return CopyMapper.ChecksumComparison.TRUE;
     }
+    LOG.error("Checksum not equal. Source checksum: {}, target checksum: {}",
+        sourceChecksum, targetChecksum);
```

Comment:
Error seems to much for this case, info should be more than enough for these cases, and I don't think we need to put the checksum in the log, rather put the ``sourcePath`` and ``targetPath``

Reply:
log checksums at debug maybe

anyway, doesn't a checksum mismatch mean "source file needs copying again". so really the thing to log is not the mismatch but why the copy is taking place